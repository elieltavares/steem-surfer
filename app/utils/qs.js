export default queryString => {
  const query = {};
  const pairs = (queryString[0] === '?'
    ? queryString.substr(1)
    : queryString
  ).split('&');

  pairs.forEach(p => {
    const pair = p.split('=');
    query[decodeURIComponent(pair[0])] = decodeURIComponent(pair[1] || '');
  });

  return query;
};
