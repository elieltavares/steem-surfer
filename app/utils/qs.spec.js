/* eslint-disable */

import qs from './qs';

describe('Query string parser', () => {
  it('(1) should parse', () => {
    const input = '?q=lorem&m=ipsum dolor sit';

    expect(qs(input)).toMatchSnapshot();
  });

  it('(2) should parse', () => {
    const input =
      '?q=t%C3%BCrk%C3%A7e%20karakterler%20ve%20bo%C5%9Fluk&created=2018';

    expect(qs(input)).toMatchSnapshot();
  });
});
