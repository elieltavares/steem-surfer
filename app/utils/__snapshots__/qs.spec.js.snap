// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query string parser (1) should parse 1`] = `
Object {
  "m": "ipsum dolor sit",
  "q": "lorem",
}
`;

exports[`Query string parser (2) should parse 1`] = `
Object {
  "created": "2018",
  "q": "türkçe karakterler ve boşluk",
}
`;
